test/solver/batch_bicgstab_kernels.cpp

```diff
@@ -224,13 +248,15 @@ TEST_F(BatchBicgstab, CanSolveLargeHpdSystem)
                                                  &logger->get_num_iterations());
     auto res_norm = gko::make_temporary_clone(exec->get_master(),
                                               &logger->get_residual_norm());
-    GKO_ASSERT_BATCH_MTX_NEAR(res.x, linear_system.exact_sol, comp_tol);
+    GKO_ASSERT_BATCH_MTX_NEAR(res.x, linear_system.exact_sol, tol * 50);
     for (size_t i = 0; i < num_batch_items; i++) {
-        auto comp_res_norm =
-            exec->copy_val_to_host(res.res_norm->get_const_values() + i);
+        auto comp_res_norm = res.res_norm->get_const_values()[i] /
+                             linear_system.rhs_norm->get_const_values()[i];
```

Comment:
I will prefer the original version because all vectors in res use the same executor

Reply:
Given that this is a struct is mainly for tests, I think it becomes easier to read and generally less error-prone if we dont have to write copy_val_to_host everywhere.

Reply:
then, maybe make the variable name clear like `host_res_norm`